open-metadata-implementation/access-services/asset-catalog/asset-catalog-server/src/main/java/org/odpi/openmetadata/accessservices/assetcatalog/listenenrs/AssetCatalogOMRSTopicListener.java

```diff
@@ -0,0 +1,209 @@
+            this.repositoryValidator = repositoryValidator;
+            this.converter = new AssetCatalogConverter(serverName, repositoryHelper);
+            this.supportedTypesForSearch = supportedTypesForSearch;
+    }
+
+    /**
+     * Unpack and deliver an instance event to the InstanceEventProcessor
+     *
+     * @param instanceEvent event to unpack
+     */
+    @Override
+    public void processInstanceEvent(OMRSInstanceEvent instanceEvent)
+    {
+        if (log.isDebugEnabled()) {
+            log.debug("Processing instance event: " + instanceEvent);
+        }
+
+        if (instanceEvent == null) {
+            log.debug("Ignored instance event - null OMRSInstanceEvent");
+            return;
+        }
+
+        OMRSEventOriginator instanceEventOriginator = instanceEvent.getEventOriginator();
+
+        if (instanceEventOriginator == null) {
+            log.debug("Ignored instance event - null OMRSEventOriginator");
+            return;
+        }
+
+        OMRSInstanceEventType instanceEventType = instanceEvent.getInstanceEventType();
+        EntityDetail entityDetail = instanceEvent.getEntity();
+        Relationship relationship = instanceEvent.getRelationship();
+
+        try{
+
+            switch (instanceEventType) {
+                case UPDATED_ENTITY_EVENT:
+                case NEW_ENTITY_EVENT:
+                case DELETED_ENTITY_EVENT:
+                case CLASSIFIED_ENTITY_EVENT:
+                case RECLASSIFIED_ENTITY_EVENT:
+                case DECLASSIFIED_ENTITY_EVENT:
+                    processEntityDetail(entityDetail);
+                    break;
+                case NEW_RELATIONSHIP_EVENT :
+                case UPDATED_RELATIONSHIP_EVENT:
+                case DELETED_RELATIONSHIP_EVENT:
+                    processRelationshipEvent(relationship);
+                    break;
+            }
+
+        } catch (Exception e) {
+            log.error("An exception occurred while processing OMRSTopic event: \n " + instanceEvent, e);
+            logExceptionToAudit(instanceEvent, e);
+        }
+
+    }
+
+    /**
+     *
+     * @param relationship the relationship to be processed
+     */
+    private void processRelationshipEvent(Relationship relationship){
+        publisher.publishEvent(relationship);
+    }
+
+    /**
+     *
+     * @param entityDetail the entityDetail to be processed
+     */
+    private void processEntityDetail(EntityDetail entityDetail){
+            String assetType = getAssetType(entityDetail);
+
+            if ( assetType != null ) {
+                AssetConverter assetConverter
+                        = new AssetConverter(entityDetail, null, repositoryHelper, serviceName, serverName);
+                Asset assetBean = assetConverter.getAssetBean();
+
+                if (assetBean == null || !this.inTheZone(assetBean.getZoneMembership())) {
+                    log.debug("Ignored instance event - Asset not in the supported zones!");
```

Comment:
We probabbly need to also do aduditLog here to stick to the pattern.

Reply:
I didn't want to load the audit log with messages that are saying it is working as expected, in case there are many events for assets not in the supported zones.